### PR TITLE
Add policy identifier label to policy API spec

### PR DESCRIPTION
**Pull request by gauravkghildiyal**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add the constant `gateway.networking.k8s.io/policy` used to identify a policy CRD to the policy related specs file. 

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/assign @robscott

### CODE REVIEW

File: apis/v1alpha2/policy_types.go [hunk 1]
```diff
@@ -16,6 +16,18 @@ limitations under the License.
 
 package v1alpha2
 
+const (
+	// PolicyLabelKey is the label whose presence identifies a CRD as a policy
+	// CRD. The value of the label SHOULD be one of the following:
```
Comment (shaneutt) [inline apis/v1alpha2/policy_types.go:21]: Are we just saying SHOULD here because we don't expect to be able to conformance test it?

Comment (robscott) [inline apis/v1alpha2/policy_types.go:21]:
```suggestion
	// PolicyLabelKey is the label whose presence identifies a CRD that
	// the Gateway API Policy attachment model. The value of the label 
	// MUST be one of the following:
```

File: apis/v1alpha2/policy_types.go [hunk 2]
```diff
@@ -16,6 +16,18 @@ limitations under the License.
...
+	//  - A label value of "inherited" indicates that this Policy is inheritable.
```
Comment (robscott) [inline apis/v1alpha2/policy_types.go:22]: I've gotten this wrong before (I think I'm also responsible for this time), but in general Kubernetes prefers for values of annotations to be capitalized. I'd recommend only specifying the upper case values in the case but maybe we can still have a case-insensitive match in tools like gwctl.

```suggestion
	//  - A label value of "Inherited" indicates that this Policy is inheritable.
```

File: apis/v1alpha2/policy_types.go [hunk 3]
```diff
@@ -16,6 +16,18 @@ limitations under the License.
...
+	//    An example of "inheritable" policy is one which if applied at the
+	//    Gateway level would affect all attached HTTPRoutes and their respective
+	//    Backends.
+	//  - A label value of "direct" indicates that the policy only affects the
```
Comment (robscott) [inline apis/v1alpha2/policy_types.go:26]:
```suggestion
	//  - A label value of "Direct" indicates that the policy only affects the
```

File: apis/v1alpha2/policy_types.go (hunk 1 above)
Comment (robscott) [reply apis/v1alpha2/policy_types.go:21]: Yeah good catch, `MUST` is probably correct here.

**Review by robscott — COMMENTED** [review]
Thanks for adding this @gauravkghildiyal! Just some nits but great to see this formally defined in our API types.

File: apis/v1alpha2/policy_types.go (hunk 2 above)
Comment (gauravkghildiyal) [reply apis/v1alpha2/policy_types.go]: Sure that sounds good. Changed.

File: apis/v1alpha2/policy_types.go (hunk 1 above)
Comment (gauravkghildiyal) [reply apis/v1alpha2/policy_types.go]: Applied. Thanks!

Comment (gauravkghildiyal) [reply apis/v1alpha2/policy_types.go]: Thanks Shane and Rob. Changed to MUST

File: apis/v1alpha2/policy_types.go (hunk 3 above)
Comment (gauravkghildiyal) [reply apis/v1alpha2/policy_types.go]: Applied

**Comment by robscott** [comment]
Thanks @gauravkghildiyal!

/lgtm
/approve

**Pull request merged**